### PR TITLE
avoid 'data_model' arguments

### LINKED ISSUE #118: adapt 2nd parameter of 'new_dm()'
Since we do not support the class `data_model` anymore, we should not require it as an argument to `dm()` and `new_dm()`

### PR DESCRIPTION
**Pull request by TSchiefer**
remove `data_model`-argument from

1. `dm()`
2. `new_dm()`

only place where the `data_model`-argument plays a larger role is in `cdm_learn_from_db()`. If we merge this PR, we need a new issue to refactor `cdm_learn_from_db()`.

fixes #118

### CODE REVIEW

**Comment by TSchiefer** [comment]
Just saw the overlaps between #118 and #69, so not sure how useful this one here is.

File: R/dm.R [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
+#' `dm()` creates a `dm` object from one or multiple [tbl] objects
+#' (tibbles or lazy data objects).
 #'
+#' @param ... Tables to add to the `dm`.  If no names are provided, the tables
+#'   are auto-named.
+#' @param .name_repair Options for name repair.
+#'   Forwarded as `repair` to [vctrs::vec_as_names()].
 #' @param src A \pkg{dplyr} table source object.
-#' @param data_model A \pkg{datamodelr} data model object, or `NULL`.
+#' @param table_names A character vector of the table names to include.
```
Comment (bbecane) [inline R/dm.R]:
```suggestion
 @param table_names A character vector of the names of the tables to include.
```

Comment (krlmlr) [reply R/dm.R]: Thanks. Please keep the `#'` .

File: R/dm.R [hunk 2]
```diff
@@ -37,118 +44,82 @@
...
-  new_dm(tables, data_model)
+  new_dm(tbls)
 })
 
 #' Low-level constructor
 #'
-#' `new_dm()` only checks if the inputs are of the correct class.
-#' If called without arguments, an empty `dm` object is created.
+#' `new_dm()` doesn't perform any checks on the input.
+#' You may need to double-check the returned object with `validate_dm()`.
```
Comment (bbecane) [inline R/dm.R]: Is  "If called without arguments, an empty `dm` object is created." not true any more?

Comment (krlmlr) [reply R/dm.R]: It is still true, but also holds for `dm()`. I wasn't sure it's worth pointing out here.

Comment (krlmlr) [reply R/dm.R]: I'll add it back.

Comment (krlmlr) [reply R/dm.R]: Done.

**Comment by krlmlr** [comment]
Thanks, @bbecane. I saw only one change, is this correct?

**Comment by bbecane** [comment]
Correct

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**